section-alert: add low progress tones (#1996)
agriculturegovau/agds-next: packages/react/src/section-alert/docs/overview.mdx

Section alert can be used to communicate different types of messages. Choosing a tone for a Section alert allows the user to understand the importance and severity of the message at a glance.

The five supported tones are:

- [success](#success)
- [error](#error)
- [warning](#warning)
- [information](#information)
- [in progress](#in-progress)

### Success

> The success section alert (green) is used for notifying the user that a task is fully completed.

Author: Colours were added back in, the low and medium tones are in their own section

```jsx live
<Stack gap={1}>
	<SectionAlert title="Your changes have been saved" tone="successHigh" />
</Stack>
```

### Error

The error section alert (red) should be used when something destructive or critical has happened.

```jsx live
<Stack gap={1}>
	<SectionAlert
		title="There was an error saving your changes"
		tone="errorHigh"
	/>
</Stack>
```

### Warning

Use warning section alerts (orange) to tell the user something urgent. Only use an alert if the information will help the user avoid a problem.

```jsx live
<Stack gap={1}>
	<SectionAlert title="A warning message for this section" tone="warningHigh" />
</Stack>
```

### Information

Use info section alerts (blue) to highlight information which is important for users.

```jsx live
<Stack gap={1}>
	<SectionAlert title="These records need more information" tone="infoHigh" />
</Stack>
```

### In progress

> Use in progress section alerts to highlight information which is relates to status.

Reviewer: ```suggestion
Use in progress section alerts to highlight information which relates to status.
```

Author: Updated in [d649ec](https://github.com/agriculturegovau/agds-next/pull/1996/commits/d649ec1f3d96f8ef4c300512d614545bf689ece5).

```jsx live
<Stack gap={1}>
	<SectionAlert title="Record is in progress" tone="inProgressLow" />
</Stack>
```

## Status and emphasis

The tone prop combines both a status and emphasis, which sets the correct importance and severity of the Section alert by automatically applying appropriate colour, iconography and accessible descriptions.

> There are nine unique tones with five levels of emphasis. All tones are available in Low emphasis but others are only available in Medium and High.

Reviewer: I notice the same paragraph in Status Badge but that one has a typo "a" instead of "are"

### High emphasis

High emphasis Section alerts use filled icons and are available in four tone colours.

Use high emphasis section alerts for:

- important system messages
- notifications that require user action

> - conveying urgent status in lists.

Reviewer: "urgent status" (singular) "in lists" (plural)?

Author: Changed to `conveying urgent statuses in lists` in [d649ec](https://github.com/agriculturegovau/agds-next/pull/1996/commits/d649ec1f3d96f8ef4c300512d614545bf689ece5).

	/>
	<SectionAlert title="A warning message for this section" tone="warningHigh" />
	<SectionAlert
		title="An information message for this section"
		tone="infoHigh"
	/>
</Stack>
```

### Medium emphasis

Medium emphasis Section alerts use stroked icons and are available in four tone colours.

Use medium emphasis section alerts to:

- convey non-urgent alerts.

```jsx live
<Stack gap={1}>
	<SectionAlert title="Your changes have been saved" tone="successMedium" />
	<SectionAlert
		title="There was an error saving your changes"
		tone="errorMedium"
	/>
	<SectionAlert
		title="A warning message for this section"
		tone="warningMedium"
	/>
	<SectionAlert
		title="An information message for this section"
		tone="infoMedium"
	/>
</Stack>
```

### Low emphasis

Low emphasis Section alerts use stroked icons and are available in the default, neutral colour only.

Use low emphasis section alerts to:

- convey non-urgent alerts
- convey progress through a process.

```jsx live
<Stack gap={1}>
	<SectionAlert title="Success" tone="successLow" />
	<SectionAlert title="Error" tone="errorLow" />
	<SectionAlert title="Warning" tone="warningLow" />
	<SectionAlert title="Info" tone="infoLow" />

> 	<Divider />

Reviewer: Why do we have this divider? And do we want to popularise the ones before it? Would they ever be used?

Author: It was part of the design docs, I think the Divider looks fine, as it splits up the progress and previous tones.

I'm not sure if we should add the ones above it (not even sure if they are that visible to alert people) but I don't think we should remove them from the docs. Should we put the progress ones above the tones?

Reviewer: I dunno. Just leave it

/>
```

## Focusing the alert

_NOTE: When focusing an alert after closing a `Drawer`, use the [focusElementOnClose prop in the Drawer component](/components/drawer#focus-custom-element-on-close)._

In the event you need to focus the alert, you can do so in one of three ways:

1. [Automatically focus on mount](#automatically-focus-on-mount).
2. [Automatically focus on update](#automatically-focus-on-update).
3. [Manually set focus](#manually-set-focus).

### Automatically focus on mount

> Press the "Toggle alert" button below to mount/unmount the alert. Notice that each time the alert is mounted it will be focused. Here we use the `focusOnMount` boolean prop to automatically handle the focusing.

Reviewer: You didn't apply my previous suggestion to this one only before moving it back

Author: Updated, sorry not sure how I missed that

> Press the "Toggle alert" button below to mount/unmount the alert. Notice that each time the alert is mounted it will be focused. Here we use the `focusOnMount` boolean prop to automatically handle the focusing.

Reviewer: ```suggestion
Press the “Toggle alert” button below to mount/unmount the alert. The alert is focused each time it is mounted. The `focusOnMount` boolean prop automatically handles the focus management.
```
